app/controllers/companies_controller.rb

```diff
@@ -89,9 +89,21 @@ def destroy
       helpers.flash_message(:alert, "#{t('companies.destroy.error')}: #{translated_errors}")
       redirect_to @company
     end
-
   end
 
+  def edit_payment
+    raise 'Unsupported request' unless request.xhr?
+    authorize Company
+
+    payment = @company.most_recent_branding_payment
+    payment.update!(payment_params) if payment
+
```

Comment:
Wouldnt that become an impossible loop? As the h-branding-fee is paid on the company page...

Reply:
It seems like a good idea, but maybe as a separate PR? In case this one is big enough already. =)

Reply:
@RobertCram - I am intentionally staying away from that aspect of licensing.  We had discussed that proposal with some SHF org people last week, but this was just a suggestion (from the dev team) of a possible way to handle non-payment.  I don't know if the board reviewed and approved this ( @thesuss ??).

Also, I personally feel that this would be an abrupt change to impose on the users until it has been communicated to them (IMO only, of course).

In any case, even if we want to do this soon my vote would be to do this as a follow-on PR.

Reply:
@patmbolger - Ok, then I should have mentioned this earlier(in the membership fee payment PR) I am not proposing a change in the way things work. I'm proposing not breaking it. At the moment a company is shown on the map as soon as an application is accepted. That used to be only after the user had paid. It's ok to do that in another PR, but it should be treated as a bug - because we are breaking existing functionality

Reply:
Just remembered: An incomplete company is not shown in listings nor on the map. 
@patmbolger isn't the H-branding fee part of what constitutes as a complete company? We decided this in one of our meetings, if it was the one with Carro and Frida? Bella agrees, so that is basically the whole board.

Reply:
Yes, that's another thing altogether.  Anyway, map visibility requires more discussion, I believe.  

For a _new_ company, making the company visible _after_ membership payment is one option.  However, SHF may want to make it visible only after membership _and_ h-branding fees are paid.

Turning visibility on/off for an existing company, which involves branding fee expiration, etc., has also got to be thought through more by SHF.

Reply:
Sorry, accidentally deleted this. here it is again: Ok, then I should have mentioned this earlier(in the membership fee payment PR) I am not proposing a change in the way things work. I'm proposing not breaking it. At the moment a company is shown on the map as soon as an application is accepted. That used to be only after the user had paid. It's ok to do that in another PR, but it should be treated as a bug - because we are breaking existing functionality

Reply:
I actually assumed it was implemented as a part of a complete company. Otherwise it does break current "process" as Robert suggests. 
In the current process they cannot edit their company (and thus display it) before they are members. 
Now without h-branding being part of a complete company they could display it all they want without ever paying the fee, and it is already too many to police.

Reply:
@thesuss Worse, they could now even do it without paying their mebership fee.

Reply:
![skarmavbild 2017-11-28 kl 14 12 24](https://user-images.githubusercontent.com/7266909/33321499-3cfa23f6-d446-11e7-81bb-2204454a3bc7.png)

Reply:
source: https://github.com/AgileVentures/shf-project/wiki/Client-meeting-about-H-Branding-licensing-and-Memberships-(notes)

Reply:
@thesuss - yes, I saw those notes.  But my impression from the meeting is that they (Board) heard our suggestion and would consider it (along with "terms of use").  I did not get the impression that we got a definitive "yes, lets go with that".  Am I wrong?

Reply:
There were and are many things in question (still) but this is not one of them. With this one it was definitive. If it was not, it is now.

Reply:
I think the confusion is/was because the bylaws indicate the fee as optional. But in reality it is not, and the bylaws will be updated after the yearly meeting in march to match reality.

Reply:
![skarmavbild 2017-11-28 kl 14 49 52](https://user-images.githubusercontent.com/7266909/33323013-701db176-d44b-11e7-8db9-c05f74e61a31.png)
Found under "SHF development plans:" in the very bottom

Reply:
OK, so can we list the business rules at this point?  Here's a starting point:

1) A user can see and edit their own company if they are a member (which implies payment of the membership fee).

2) A company will be made _visible_ to others (show on map, appear in lists, be searchable, etc.) if the h-branding fee has been paid.  Otherwise, the company is **not** _visible_.

3) A company for which the branding period has expired is turned _invisible_ upon expiration (no grace period, no admin switch required to turn it on or off).

4) Other than the rules above, no other criteria effect company visibility (that is, a company can remain _visible_ even though the the company's members have all expired their membership period).

5) A user whose membership has expired can still operate on the site as a member (edit their company, pay fees, etc.) until and if the admin revoked their membership status.

Reply:
1) Yes
2) And is otherwise complete 
3) Yes
4) I'd say if a company has no members (which it doesnt if they havent paid) it should not be visible
5) For this I'd say the same, if the membership has expired they are de facto not members anymore, but revert to users with an approved application.

Reply:
2. what is meant by "complete" company?
4. that situation is an existing company which has members but those members have not yet renewed their membership application(s).
5. The described behavior is what we've shown people up to now - the idea is that the "member" is given a grace period (and a reminder from the admin or automated email later) and can still login and pay the fee (revoking membership status means they will only see the visitor menu).  I'd recommend keeping this as is until we get some operating experience with the new structure.

Reply:
(had to edit the above since markdown turned my numbers into sequential numbers)

Reply:
- 1 - Well there is code in place now that only lists complete companies. That + h-branding = the new complete. 
- 4 - yes
- 5 - Okay, sure.

Reply:
This discussion is great and much needed (although not sure for this PR, because it is new functionality). I just want to make clear that my original remark was just asking a small change (moving a bit of code) to prevent breaking functionality, i.e. (disregarding any H-branding stuff) a user can now make a company "complete" as soon as an application is accepted. That should only be possible after the necessary payments (at least a membership fee) have been done.

Reply:
@RobertCram - yes, great discussion and clarifies things.  I will look into how we can prevent the breakage you mentioned.  (I'll also post these business rules on the wiki).

Reply:
All good stuff!  These are definitely sketches of some _stories / use cases._  I volunteer to write them down so we can all think about them.  I'll let you all know where I put them.  (probably PT if they are well defined; wiki (developer meeting agenda items? if not)